### ISSUE TITLE
Screenshots fail to display in Flathub page for `net.sourceforge.pdfchain` app

### ISSUE THREAD

**Issue by nbenitez** [ximion/appstream#686]
I initially reported this bug to [Flathub website component](https://github.com/flathub-infra/website/issues/4192), but [was told there](https://github.com/flathub-infra/website/issues/4192#issuecomment-2495738270) that this is really an appstream bug, so I'm filing it here too. 

--

I'm the maintainer of https://flathub.org/apps/net.sourceforge.pdfchain and I just noticed that although the app has [6 different screenshots](https://github.com/flathub/net.sourceforge.pdfchain/blob/0a65c5127c9133418c09d6ae00d648a9c7644223/net.sourceforge.pdfchain.metainfo.xml#L45-L68), the webpage only displays one screenshot but repeated 6 times.

I presume the bug is related to the fact that [screenshots have same name](https://github.com/flathub/net.sourceforge.pdfchain/blob/0a65c5127c9133418c09d6ae00d648a9c7644223/net.sourceforge.pdfchain.metainfo.xml#L45-L68) although their paths are different. They are different URI's so I think they should be permitted like that.

**Comment by ximion** [comment]
Do you have a way to reproduce this? The explanation doesn't make much sense, because `appstreamcli compose` saves screenshots in a directory with a hash, and renames downloaded images by their node ID (so screenshot 1 geta "image-1.png" etc.).
I couldn't reproduce this here, but I also tried it with a different file and not your exact example, so there may be something else going on...

**Comment by nbenitez** [comment]
I re-uploaded screenshots with different names to fix it, by you can see in git history [how they look when they were problematic](https://github.com/flathub/net.sourceforge.pdfchain/blob/d250f37884340dc2e22df58739312eed7344d471/net.sourceforge.pdfchain.metainfo.xml#L45-L69).

**Comment by ximion** [comment]
Yes, I used that block and couldn't reproduce the issue - compose generates screenshots like this:
```
image-1_1248x752@1.png  image-2_224x135@1.png   image-3_624x376@1.png   image-4_752x453@1.png   image-5_orig.png        image-7_1248x752@1.png
image-1_224x135@1.png   image-2_624x376@1.png   image-3_752x453@1.png   image-4_orig.png        image-6_1248x752@1.png  image-7_224x135@1.png
image-1_624x376@1.png   image-2_752x453@1.png   image-3_orig.png        image-5_1248x752@1.png  image-6_224x135@1.png   image-7_624x376@1.png
image-1_752x453@1.png   image-2_orig.png        image-4_1248x752@1.png  image-5_224x135@1.png   image-6_624x376@1.png   image-7_752x453@1.png
image-1_orig.png        image-3_1248x752@1.png  image-4_224x135@1.png   image-5_624x376@1.png   image-6_752x453@1.png   image-7_orig.png
image-2_1248x752@1.png  image-3_224x135@1.png   image-4_624x376@1.png   image-5_752x453@1.png   image-6_orig.png
```

It will not use any element from the URL, and I can't see how this issue would ever happen (I processed your original screenshot data...).
Unfortunately the Flathub issue also didn't have any details so I could find out if anything was wrong on their side. The images served from their website are webp images, which compose doesn't output (yet...), so they must have some postprocessing.

I'll close this for now, but please feel free to reopen if this issue persists or can somehow be reproduced!

**Comment by razzeee** [comment]
```
find -L /tank/repo/$REPONAME/media -type f -iname '*.jpg' -or -iname '*.png' | parallel '/usr/local/bin/webp-convert {}' || true
find -L /tank/repo/$REPONAME/appstream -type f -iname '*.jpg' -or -iname '*.png' | parallel --progress 'cwebp -quiet -q 100 -mt {} -o {.}.webp' || true

```

Is what's used to create the webp files

**Comment by ximion** [comment]
That shouldn't cause *any* problems, you're not even editing the generated XML... This issue is really strange!

**Comment by razzeee** [comment]
Yeah, we change the path in the websites backend, can't risk g-s or discover getting webp all of a sudden

**Comment by ximion** [comment]
I have a version that uses JPEG-XL for icons, which is super nice (much smaller than PNG while also lossless) as well as screenshots (using lossy compression there).
But since Flathub exists, there's little point in merging it until browsers have wide support (and software centers do as well).
At least compose runs optipng to get sizes down (and if JXL doesn't tun out to be the final solution, adding WebP as format to support for clients is an option too).

**Comment by razzeee** [comment]
Yeah, jpeg xl will be nice, once browser support eventually works out, especially the loading behavior. Until that's the case webp is the way to go.

We need to use the lossless compression for icons though, but I have no access to that server.

Is jpeg-xl for screenshots not something you think would be nice?
